### PR TITLE
Bugfix: Reset passphrase before using

### PR DESCRIPTION
**Pull request by ghost**
https://app.clickup.com/t/37308523/CHI-2009

### CODE REVIEW

**Comment by amanCypherock** [comment]
Should add a check if a passphrase is used in the wallet or not when converting from mnemonic to seed. Based on that we should either pass a passphrase or an empty string.

**Comment by amanCypherock** [comment]
Also, clear the passphrase in all flows on flow resets, if not done already

File: src/level_four/core/controller/receive_transaction_controller.c [hunk 1]
```diff
@@ -75,6 +75,7 @@ void receive_transaction_controller()
     switch (flow_level.level_three) {
 
     case RECV_TXN_FIND_XPUB: {
+        memzero(wallet_credential_data.passphrase, sizeof(wallet_credential_data.passphrase));
```
Comment (ujjwal-cyph) [inline src/level_four/core/controller/receive_transaction_controller.c:78]: lets reset this in `reset_flow_level` as well.

Comment (ujjwal-cyph) [reply src/level_four/core/controller/receive_transaction_controller.c:78]: Also, add this just after passphrase is used. i.e., after the call to `mnemonic_to_seed`

**Review by ujjwal-cyph — APPROVED** [review]

File: src/controller_main.c [hunk 2]
```diff
@@ -229,6 +229,8 @@ void reset_flow_level()
 
     //clear memory
     memzero(wallet.password_double_hash, sizeof(wallet.password_double_hash));
+    memzero(wallet_credential_data.passphrase, sizeof(wallet_credential_data.passphrase));
+
```
Comment (ujjwal-cyph) [inline src/controller_main.c:233]: no space. `cy_free` belongs to same action group i.e., cleaning memory allocations

Comment (ujjwal-cyph) [inline src/controller_main.c:233]:
```suggestion
```

**Pull request merged**